src/Microsoft.DotNet.Cli.Utils/ProjectContextExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+using System.IO;
+using Microsoft.DotNet.ProjectModel;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal static class ProjectContextExtensions
+    {
+        public static string ProjectName(this ProjectContext context) => context.RootProject.Identity.Name;
+
+        public static string GetOutputPath(this ProjectContext context, string configuration, string currentOutputPath)
+        {
+            var outputPath = string.Empty;
+
+            if (string.IsNullOrEmpty(currentOutputPath))
+            {
+                outputPath = Path.Combine(
+                    GetDefaultRootOutputPath(context, currentOutputPath),
+                    Constants.BinDirectoryName,
+                    configuration,
+                    context.TargetFramework.GetTwoDigitShortFolderName());
+            }
+            else
+            {
+                outputPath = currentOutputPath;
+            }
+
+            return outputPath;
+        }
+
+        public static string GetIntermediateOutputPath(this ProjectContext context, string configuration, string intermediateOutputValue, string currentOutputPath)
+        {
+            var intermediateOutputPath = string.Empty;
```

Comment:
That would be a way of doing. However, there currently is ambiguity between Compile's GetOutputPath (which I moved into the new ProjectContextExtensions and ProjectContext's own GetOutputDirectoryPath).

I suggest opening a new issue about this and not including the change in this PR (which tries to faithfully presever Compile's semantics)